src/main/java/uk/gov/hmcts/darts/cache/token/service/TokenRegisterable.java

```diff
@@ -6,21 +6,75 @@
 
 import java.util.Optional;
 
+/**
+ * This is a token registerable implementation (effectively a cache implementation) that allows us to set and get
+ * values {@link uk.gov.hmcts.darts.cache.token.service.value.CacheValue} from the cache.
+ */
 public interface TokenRegisterable {
 
-    public static final String CACHE_PREFIX = "darts-gateway-token-service";
+    String CACHE_PREFIX = "darts-gateway-token-service";
 
-    Optional<Token> store(CacheValue value) throws CacheException;
+    /**
+     * Stores a cache value into the cache.
+     * @return The token in which the cache value is mapped against
+     */
+    Optional<Token> store(CacheValue value);
 
-    Optional<Token> store(CacheValue value, Boolean reuseTokenIfPossible) throws CacheException;
+    /**
+     * Stores a cache value into the store. If the cache value is already mapped into the cache we can
+     * return the same id if we choose to.
+     * This method should ensure the token is valid
+     * (whatever that may mean) before it is returned. Clients can validate the tokens at any time thereafter
+     * by calling {@link Token#valid()}
+     * @param reuseTokenIfPossible If set to true we reuse the token else we create a new one
+     */
+    Optional<Token> store(CacheValue value, Boolean reuseTokenIfPossible);
 
+    /**
+     * This method takes the service context and works out whether it needs to generate a new token ot not.
+     * If a token does exists but it is invalid then a new token is expected to be generated
+     *
```

Comment:
Seems to be missing the other `@param` if we need the javadoc

Reply:
Good spot. Thought checkstyle would pick this up